iothub/device/src/DeviceClient.cs

```diff
@@ -14,6 +14,7 @@ namespace Microsoft.Azure.Devices.Client
     /// <summary>
     /// Contains methods that a device can use to send messages to and receive from the service.
     /// </summary>
+    /// <threadsafety static="true" instance="true" />
```

Comment:
A customer requested to document this. I will have to watch out for the msdn docs after our next release to see if this works as expected. Also appreciate comments if this is not true. In my understanding, the DeviceClient is thread-safe.